src/electron/index.ts

```diff
@@ -334,18 +340,57 @@ function sendConnectionStatus(status: ConnectionStatus, connectionId: string) {
   } else {
     console.warn(`received ${event} event but no mainWindow to notify`);
   }
+  createTrayIcon(status);
 }
 
+// Connects to the specified server, if that server is reachable and the credentials are valid.
 promiseIpc.on(
-    'start-proxying', (args: {config: cordova.plugins.outline.ServerConfig, id: string}) => {
-      return startVpn(args.config, args.id).catch((e) => {
-        console.error(`could not connect: ${e.name} (${e.message})`);
-        throw errors.toErrorCode(e);
-      });
+    'start-proxying', async (args: {config: cordova.plugins.outline.ServerConfig, id: string}) => {
+      // yes, it's weird the UI doesn't do this.
```

Comment:
Okay, now I get it :-) I chaned this to a TODO - as discussed, this will be a big (albeit straightforward) change and this is no worse than what we have today.